transaction/redoTx.go

```diff
@@ -489,11 +484,12 @@ func (t *redoTx) Begin() error {
 }
 
 /* Persists the update written to redoLog during the transaction lifetime. For
- * nested transactions, End() call to inner transaction does nothing.
+ * nested transactions, End() call to inner transaction does nothing. Returns a
+ * bool indicating if it is safe to release the transaction handle.
  */
-func (t *redoTx) End() error {
+func (t *redoTx) End() bool {
 	if t.level == 0 {
-		return errors.New("[redoTx] End: no transaction to commit")
+		return true
```

Comment:
Should we throw an error here instead as hitting this case indicates a programming error?

Reply:
Yes. Since we are not returning an error now, we would have to crash the program in this case. I decided against that since it is not a "fatal" error. But you are right, we can totally crash here instead.

Reply:
I agree, this is not a fatal error. How about adding a `log.Printf()` so that this error is logged to `stderr`.